gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -185,9 +185,39 @@ public void handleUnpublishedWorkUnits(Collection<? extends WorkUnitState> state
     return datasetRoots;
   }
 
+  /**
+   * Unlike other preserving attributes of files (ownership, group, etc.), which is preserved in writer,
+   * some of the attributes have to be set during publish phase like ModTime,
+   * and versionStrategy (usually relevant to modTime as well), since they are subject to change with Publish(rename)
+   */
+  private void preserveFileAttrInPublisher(CopyableFile copyableFile) throws IOException {
+    // Preserving File ModTime, keep the accessing time of the target file.
+    if (copyableFile.getPreserve().preserve(PreserveAttributes.Option.MOD_TIME)) {
+      fs.setTimes(copyableFile.getDestination(), copyableFile.getOriginTimestamp(),
+          fs.getFileStatus(copyableFile.getDestination()).getAccessTime());
```

Comment:
Why not add accessTime as a field in CopyableFile and populate it when CopyableFiles are computed? We can avoid the fs.getFileStatus() RPC call here. Also, not sure whether we should preserve the file "accessTime" from the source. We might be better off setting this to -1.

Reply:
Actually I think a bit more into the accessTime, it might makes more sense to set this to -1 since the preservation of ModTime is actually breaking the semantics already (we preserve it for special use cases only) and it doesn't make sense to keep the original access time either.